camel/agents/chat_agent.py

```diff
@@ -96,7 +96,7 @@ class ChatAgent(BaseAgent):
             (default: :obj:`None`)
         output_language (str, optional): The language to be output by the
             agent. (default: :obj:`None`)
-        function_list (List[OpenAIFunction], optional): List of available
+        tools (List[OpenAIFunction], optional): List of available
```

Comment:
I'm adding a new abstract type [BaseToolkit](https://github.com/camel-ai/camel/blob/cef1bcb308420d98028a55021b322fb7bc93a8aa/camel/toolkits/base.py) and [GithubToolkit](https://github.com/camel-ai/camel/blob/cef1bcb308420d98028a55021b322fb7bc93a8aa/camel/toolkits/github_toolkit.py) in another PR.

Do you think it would make sense to accept an instance of `BaseTookit` for this parameter?

Reply:
Hey @rsrbk , thanks for the comment, I think we need uniformed instance to manage tools/functions, why we introduce toolkit, could you help me understand the key benefits for having this additional toolkit instance?

Reply:
Mostly because we saw that having a class instance might be beneficial for certain tools. In case of github, you would have to create a new Github object and access your token every time you call a function.